### PR TITLE
Syntax highlighting: metadata and more keywords.

### PR DESCRIPTION
**Pull request by eltoder**
Add `enum` keyword, fix syntax for `import` and `export` statements, move
`factory` to storage class for consistency, and highlight @metadata
annotations.

### CODE REVIEW

File: syntax/dart.vim [hunk 1]
```diff
@@ -37,7 +37,12 @@ syntax match   dartUserLabelRef   "\k\+" contained
 syntax region  dartLabelRegion   transparent matchgroup=dartLabel start="\<case\>" matchgroup=NONE end=":"
 syntax keyword dartLabel         default
 
-syntax match dartLibrary         "^\(import\|part of\|part\|export\|library\|show\|hide\)\s"
+syntax keyword dartLibrary       import export nextgroup=dartUri skipwhite
+syntax region  dartUri           start=+r\=\z(["']\)+ end=+\z1+ nextgroup=dartCombinator skipwhite
+syntax keyword dartCombinator    contained show hide deferred
+syntax match   dartLibrary       "\<\(library\|part of\|part\)\>"
```
Comment (cbracken) [inline syntax/dart.vim]: Keep the `^` -- `library`, `part of`, `part` are keywords but not reserved words, so it's legal to use them as identifiers.

File: syntax/dart.vim [hunk 2]
```diff
@@ -37,7 +37,12 @@ syntax match   dartUserLabelRef   "\k\+" contained
 syntax region  dartLabelRegion   transparent matchgroup=dartLabel start="\<case\>" matchgroup=NONE end=":"
 syntax keyword dartLabel         default
 
-syntax match dartLibrary         "^\(import\|part of\|part\|export\|library\|show\|hide\)\s"
+syntax keyword dartLibrary       import export nextgroup=dartUri skipwhite
```
Comment (cbracken) [inline syntax/dart.vim]: Keep the `^`. `import`, `export` are keywords, but not reserved words in Dart, and can be used as identifiers.

**Comment by cbracken** [comment]
Other than two minor nits which are easily fixed, looks great! Thanks for sending this!

**Comment by eltoder** [comment]
Thanks! Done both. Also changed a bit to allow matching multiple combinators for each import, as per language spec.

**Comment by cbracken** [comment]
lgtm!

**Pull request merged**